ampd/src/queue/msg_queue.rs

```diff
@@ -9,7 +9,10 @@ pub struct MsgQueue {
 impl MsgQueue {
     pub fn push(&mut self, msg: Any, gas_cost: Gas) {
         self.msgs.push(msg);
-        self.gas_cost += gas_cost;
+        self.gas_cost = self
+            .gas_cost
+            .checked_add(gas_cost)
+            .expect("arithmetic operation overflows");
     }
```

Comment:
I changed it to throw an error instead of panic, however I don't really like changing the signature to consume and return self, I think it's counterintuitive since usually other structs with a function similar to `push()` like Vec and HashMap are expected to be mutable to add new elements. I also think it's not necessary since we are not chaining push to some other functions

Reply:
even with a fluent interface the struct is mutable, we wouldn't create an immutable copy. Checking the current code, we use loops over pushes in multiple places, the reason there is no chaining is that we currently can't chain because it's interface doesn't allow it. A fluent interface would also allow to use it with `fold()`-like functions

Reply:
I am OK with `push` consuming self and therefore enabling chaining, but I am not sure what difference panic v.s. returning error here makes?

Reply:
could we leave the refactoring to a fluent interface struct out of scope of this PR?